blockmanager.go

```diff
@@ -912,16 +989,47 @@ func (c *checkpointedCFHeadersQuery) handleResponse(req, resp wire.Message,
 	select {
 	case c.headerChan <- r:
 	case <-c.blockMgr.quit:
-		return query.Progress{
-			Finished:   false,
-			Progressed: false,
-		}
+		return query.NoResponse
 	}
 
-	return query.Progress{
-		Finished:   true,
-		Progressed: true,
+	return query.Finished
+}
+
+// sendQueryMessageWithEncoding sends a message to the peer with encoding.
+func sendQueryMessageWithEncoding(peer query.Peer, req query.ReqMessage) error {
```

Comment:
Because it follows this function signature: 
https://github.com/Chinwendu20/neutrino/blob/64b278771ff75da0b30136af7d4ab0ace06d897e/query/interface.go#L169-L171
And since this mean't to be called in the query package it would cause a cyclic dependence.

Reply:
It looks like `QueueMessageWithEncoding` is also part of the `Peer` interface, so why do we need this ServerPeer downcast?

Reply:
That was removed in this commit: 9ab1d8b82d5fd5c8c73fe66a59b8dc6b991787